src/SFA.DAS.EmployerFinance/Models/Account.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SFA.DAS.EmployerFinance.Models
+{
+    public class Account : Entity
+    {
+        public long Id { get; internal set; }
+        public string HashedId { get; internal set; }
+        public string PublicHashedId { get; internal set; }
+        public string Name { get; internal set; }
+        public DateTime Created { get; internal set; }
+        public DateTime? Updated { get; internal set; }
+        public IEnumerable<AccountPayeScheme> AccountPayeSchemes => _accountPayeSchemes;
+        
+        internal readonly List<AccountPayeScheme> _accountPayeSchemes = new List<AccountPayeScheme>();
+
+        public Account(long id, string hashedId, string publicHashedId, string name, DateTime created)
+        {
+            Id = id;
+            HashedId = hashedId;
+            PublicHashedId = publicHashedId;
+            Name = name;
+            Created = created;
+        }
+
+        private Account()
+        {
+        }
+        
+        public void UpdateName(string name, DateTime updated)
+        {
+            if (IsUpdatedNameDateChronological(updated) && IsUpdatedNameDifferent(name))
+            {
+                Name = name;
+                Updated = updated;
+            }
+        }
+
+        public AccountPayeScheme AddPayeScheme(string employerReferenceNumber, DateTime created)
+        {
+            EnsurePayeSchemeHasNotAlreadyBeenAdded(employerReferenceNumber);
+            
+            var accountPayeScheme = new AccountPayeScheme(Id, employerReferenceNumber, created);
+            
+            _accountPayeSchemes.Add(accountPayeScheme);
+
+            return accountPayeScheme;
+        }
+
+        public void RemovePayeScheme(AccountPayeScheme accountPayeScheme, DateTime removed)
+        {
+            EnsureAccountPayeSchemeHasBeenAdded(accountPayeScheme);
```

Comment:
I think we'll need chronological checks here unless we can use the surrogate ID from employer accounts.

Reply:
See reply to previous comment.